app/graph/types/team_type.rb

```diff
@@ -400,4 +400,56 @@ def statistics(period:, language: nil, platform: nil)
     return nil unless User.current&.is_admin
     TeamStatistics.new(object, period, language, platform)
   end
+
+  field :bot_query, [TiplineSearchResultType], null: true do
+    argument :search_text, GraphQL::Types::String, required: true
+  end
+
+  def bot_query(search_text:)
+    unless User.current&.is_admin
+      raise GraphQL::ExecutionError, "You do not have permission to perform this action"
+    end
```

Comment:
Suggestion: Make this logic consistent with this one: https://github.com/meedan/check-api/blob/develop/app/graph/types/team_type.rb#L400. I suggest to implement a helper method that returns `nil` if the user is not a super-admin, and use this method here and above. This method could be defined in the base controller: https://github.com/meedan/check-api/blob/develop/app/controllers/api/v1/graphql_controller.rb.

Reply:
It seems that methods in the GrapqlController are not exposed to `TeamType`. I will continue checking this and implement it in the next PR for the actual implementation.

But I made the check consistent with the one above.